File: client/java/brownshome/scriptwars/connection/ConnectionException.java
package brownshome.scriptwars.connection;

public class ConnectionException extends Exception {
	ConnectionStatus connectionStatus;
	
	public ConnectionException(ConnectionStatus connectionStatus, Throwable t) {
		super(t);
		
		this.connectionStatus = connectionStatus;
	}
	
	public ConnectionException(ConnectionStatus connectionStatus) {
		this.connectionStatus = connectionStatus;
	}

	public ConnectionStatus getConnectionStatus() {
		return connectionStatus;
	}
}

File: client/csharp/Script Wars Client/Script Wars Client/ScriptWars/Connection/ConnectionException.cs
using System;

namespace ScriptWars.Connection
{
    public class ConnectionException : Exception
    {
        public ConnectionStatus ConnectionStatus { get; }

        public ConnectionException(ConnectionStatus connectionStatus, Exception innerException) :
            base(connectionStatus.ToString(), innerException)
        {
            ConnectionStatus = connectionStatus;
        }

        public ConnectionException(ConnectionStatus connectionStatus, string message = null) : base(
            message ?? connectionStatus.ToString())
        {
            ConnectionStatus = connectionStatus;
        }
    }
}